File: src/SIStatisticsService.Contract/Models/PackageStats.cs
namespace SIStatisticsService.Contract.Models;

/// <summary>
/// Represents statistical data for a package.
/// </summary>
/// <param name="TopLevelStats">Top-level statistical data for the package.</param>
/// <param name="QuestionStats">A dictionary containing statistics for each question in the package.</param>
public sealed record PackageStats(
    PackageTopLevelStats TopLevelStats,
    Dictionary<string, QuestionStats> QuestionStats);

File: web/src/models/PackageStats.ts
import PackageTopLevelStats from './PackageTopLevelStats';
import QuestionStats from './QuestionStats';

/** Represents statistical data for a package. */
export default interface PackageStats {
	/** Top-level statistical data for the package. */
	topLevelStats: PackageTopLevelStats;

	/** A dictionary containing statistics for each question in the package. */
	questionStats: Record<string, QuestionStats>;
}
